app/routes/products_.$productId/route.tsx

```diff
@@ -31,6 +31,7 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
 export default function ProductDetailsPage() {
     const { product } = useLoaderData<typeof loader>();
     const { setIsOpen } = useCartOpen();
+    const [submitClicked, setSubmitClicked] = useState(false);
```

Comment:
submitClicked - it's not really a submit, because especially in Remix submit is about form submission.
maybe
`addToCartAttempted`
`addToCartClicked`
? what do you think ?

Reply:
renamed to addToCartAttempted